src/code.cloudfoundry.org/loggregator/plumbing/conversion/tov2.go

```diff
@@ -56,19 +56,15 @@ func ToV2(e *events.Envelope, usePreferredTags bool) *v2.Envelope {
 // TODO: Do we still need to do an interface?
 func setV2Tag(e *v2.Envelope, key string, value interface{}, usePreferredTags bool) {
 	if usePreferredTags {
-		switch value.(type) {
-		case string:
-			e.GetTags()[key] = value.(string)
-		case int32, int64:
-			e.GetTags()[key] = fmt.Sprintf("%d")
-		case float64:
-			e.GetTags()[key] = fmt.Sprintf("%f")
-		default:
-			e.GetTags()[key] = fmt.Sprintf("%v", value)
+		if s, ok := value.(string); ok {
+			e.GetTags()[key] = s
+			return
 		}
 
+		e.GetTags()[key] = fmt.Sprintf("%v", value)
```

Comment:
I'm cool with this. It means that if someone calls `setV2Tag` with weird values they will be outputted the way Go sees them. Not sure that is much of a worry though.